### DISCUSSION TITLE
extend Cut

### DISCUSSION THREAD

**Discussion by tangentstorm** [jsoftware/jsource#247] [Ideas]
I'd like to propose two extensions for the Cut (`;.`) dyad.

Some partitioned algorithms require to have position of non-complete shard at the begin of array. Current u;.3 and u;._3 implementation provides that shard at the end of array only. Following is the model of extension desired ("5" is selected randomly for example only):

```
   (4 ,: 3) <;.5 (i. 10)
┌───┬─────┬─────┐
│0 1│2 3 4│6 7 8│
└───┴─────┴─────┘
```
Sometimes there is a need to align selected shard within frame not to the left (beginning) of the frame as currently implemented, but to the right (end), too:

```
   (_4 ,: 3) <;.3 (i. 10)
┌─────┬─────┬───┐
│1 2 3│5 6 7│8 9│
└─────┴─────┴───┘
   (_4 ,: 3) <;.5 (i. 10)
┌───┬─────┬─────┐
│0 1│3 4 5│7 8 9│
└───┴─────┴─────┘
```

-- Igor Zhuravlov <2009-12-26T16:55:18Z)>

**Comment by jip** [comment]
Cross-reference: [[JWiki] System/Interpreter/Requests](https://code.jsoftware.com/wiki/System/Interpreter/Requests#extend_Cut), Igor Zhuravlov, Sat Dec 26 16:55:18 UTC 2009.

**Comment by HenryHRich** [comment]
Good idea IMO
